src/main/java/org/terasology/lightandshadowresources/logic/domino/DominoSystem.java

```diff
@@ -0,0 +1,230 @@
+        if (targetBlockComp == null) {
+            event.consume();
+            return;
+        }
+
+        Vector3f horizDir =
+                new Vector3f(event.getDirection())
+                        .setComponent(1, 0); // set y dimension to 0
+        Side facingDir = Side.inDirection(horizDir);
+        if (!facingDir.isHorizontal()) {
+            event.consume();
+            return;
+        }
+
+        Vector3ic blockPos = targetBlockComp.getPosition();
+        Side offsetDir = facingDir.reverse();
+        Vector3i primePos = blockPos.add(offsetDir.direction(), new Vector3i());
+        Block primeBlock = worldProvider.getBlock(primePos);
+        if (!primeBlock.isReplacementAllowed()) {
+            event.consume();
+            return;
+        }
+
+        Block belowBlock = worldProvider.getBlock(primePos.x, primePos.y - 1, primePos.z);
+        Block aboveBlock = worldProvider.getBlock(primePos.x, primePos.y + 1, primePos.z);
+
+        // Determine top and bottom blocks
+        Vector3i bottomBlockPos = new Vector3i();
+        Vector3i topBlockPos = new Vector3i();
+        if (belowBlock.isReplacementAllowed()) {
+            bottomBlockPos.set(primePos.x, primePos.y - 1, primePos.z);
+            topBlockPos.set(primePos);
+        } else if (aboveBlock.isReplacementAllowed()) {
+            bottomBlockPos.set(primePos);
+            topBlockPos.set(primePos.x, primePos.y + 1, primePos.z);
+        } else {
+            event.consume();
+            return;
+        }
+        Side attachSide = determineAttachSide(facingDir, offsetDir, bottomBlockPos, topBlockPos);
+        if (attachSide == null) {
+            event.consume();
+            return;
+        }
+
+        Side risenSide = facingDir.reverse();
+
+        Block newBottomBlock = domino.bottomBlockFamily.getBlockForPlacement(new BlockPlacementData(bottomBlockPos,
+                risenSide, TOP));
+        Block newTopBlock = domino.topBlockFamily.getBlockForPlacement(new BlockPlacementData(bottomBlockPos,
+                risenSide, TOP));
+
+        Map<org.joml.Vector3i, Block> blockMap = new HashMap<>();
+        blockMap.put(bottomBlockPos, newBottomBlock);
+        blockMap.put(topBlockPos, newTopBlock);
+        PlaceBlocks blockEvent = new PlaceBlocks(blockMap, event.getInstigator());
+        worldProvider.getWorldEntity().send(blockEvent);
+
+        if (!blockEvent.isConsumed()) {
+            EntityRef newDomino = entityManager.create(domino.dominoRegionPrefab);
+            entity.removeComponent(MeshComponent.class);
+
+            newDomino.addComponent(new BlockRegionComponent(new BlockRegion(bottomBlockPos).union(topBlockPos)));
+
+            Vector3fc dominoCenter = new Vector3f(bottomBlockPos).add(0, 0.5f, 0);
+            newDomino.addComponent(new LocationComponent(dominoCenter));
+
+            DominoComponent newDominoComp = new Domino.getComponent(DominoComponent.class);
```

Comment:
There is no `Domino` class, hence the compiler doesn't understand `new Domino`.
What exactly is your intention here? Do you want to create a new `DominoComponent`?
If so, you probably want to do `new DominoComponent` instead.

Reply:
Yeah, I want to create a new DominoComponent and use it to make the domino fall and rise. To be honest, I copied the code from the door"s logic from CoreAdvancedaAssets and then modified it (refactored variables and some fields).